src/MetaCompilation/MetaCompilation/MetaCompilation/DiagnosticAnalyzer.cs

```diff
@@ -1673,32 +1668,33 @@ private SyntaxToken SpanAnalysis(SyntaxToken startToken, SyntaxToken endToken, S
                 }
 
                 var endArgIdentifier = endArg.Expression as IdentifierNameSyntax;
-                if (endArgIdentifier == null || endArgIdentifier.Identifier.Text != endToken.Text)
+                if (endArgIdentifier == null)
                 {
                     return emptyResult;
                 }
 
-                return spanToken;
+                emptyResult.Add(startArgIdentifier.Identifier.Text);
+                emptyResult.Add(endArgIdentifier.Identifier.Text);
+                return emptyResult;
             }
 
-            // Checks the creation of the diagnostics location, returns a SyntaxToken of "" if analysis failed
-            private SyntaxToken LocationAnalysis(SyntaxToken statementIdentifierToken, SyntaxToken spanToken, SyntaxList<StatementSyntax> statements)
+            // Checks the creation of the diagnostics location, returns a string of "" if analysis failed, else returns name of span variable
+            private string LocationAnalysis(SyntaxToken statementIdentifierToken, string locationName, StatementSyntax statement)
             {
-                var emptyResult = SyntaxFactory.Identifier("");
-
-                var statement = statements[7] as LocalDeclarationStatementSyntax;
-                if (statement == null)
+                string emptyResult = "";
```

Comment:
@jepetty Since strings are nullable, it probably makes more sense to have emptyResults for strings be null throughout